src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Dumper;
+
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\Marking;
+
+/**
+ * PlantUmlDumper dumps a workflow as a PlantUML file.
+ *
+ * You can convert the generated dot file with the plantuml.jar utility (http://plantuml.com/):
+ *
+ * php bin/console workflow:dump pull_request travis --format=puml | java -jar plantuml.jar -p  > workflow.png
+ *
+ * @author Sébastien Morel <morel.seb@gmail.com>
+ */
+class PlantUmlDumper implements DumperInterface
+{
+    const INITIAL = 'initial';
+
+    const MARKED = 'marked';
+
+    const SYMFONY_LOGO = 'sprite $sf_logo [81x20/16z] {
+hPNRaYiX24K1xwBo_tyx6-qaCtDEJ-KXLYMTLbp0HWcHZr3KRDJ8z94HG3jZn4_mijbQ2ryJoFePtXLWA_qxyGy19DpdY_10z11ZAbGjFHRwcEbcKx5-wqsV
+yIMo8StMCHKh8ZUxnEwrZiwRAUOvy1lLcPQF4lEFAjhzMd5WOAqvKflS0Enx8PbihiSYXM8ClGVAseIWTAjCgVSAcnYbQG79xKFsZ0VnDCNc7AVBoPSMcTsX
+UnrujbYjjz0NnsObkTgnmolqJD4QgGUYTQiNe8eIjtx4b6Vv8nPGpncn3NJ8Geo9W9VW2wGACm_JzgIO8A8KXr2jUBCVGEAAJSZ6JUlsNnmOzmIYti9G7bjL
+8InaHM9G40NkwTG7OxrggvNIejA8AZuqyWjOzTIKi-wwYvjeHYesSWuPiTGDN5THzkYLU4MD5r2_0PDhG7LIUG33z5HtM6CP3icyWEVOS61sD_2ZsBfJdbVA
+qM53XHDUwhY0TAwPug3OG9NonRFhO8ynF3I4unuAMDHmSrXH57V1RGvl9jafuZF9ZhqjWOEh98y0tUYGsUxkBSllIyBdT2oM5Fn2-ut-fzsq_cQNuL6Uvwqr
+knh4RrvOKzxZfLV3s0rs_R_1SdYt3VxeQ1_y2_W2
+}';
+
+    const DEFAULT_TRANSITION_TYPE = 'arrow';
+
+    const SQUARE_TRANSITION_TYPE = 'square';
+
+    protected static $defaultOptions = array(
+        'skinparams' => array(
+            'titleBorderRoundCorner' => 15,
+            'titleBorderThickness' => 2,
+            'state' => array(
+                'BackgroundColor<<'.self::INITIAL.'>>' => '#87b741',
+                'BackgroundColor<<'.self::MARKED.'>>' => '#3887C6',
+                'BorderColor' => '#3887C6',
+                'BorderColor<<'.self::MARKED.'>>' => 'Black',
+                'FontColor<<'.self::MARKED.'>>' => 'White',
+            ),
+            'agent' => array(
+                'BackgroundColor' => '#ffffff',
+                'BorderColor' => '#3887C6',
+            ),
+        ),
+    );
+
+    protected $transitionType = self::DEFAULT_TRANSITION_TYPE;
```

Comment:
move everything protected to private

Reply:
I still don't understand that. I understand the concept of not allowing inheritance first. But for a dumper, I don't see why it would be bad to extend it and tweak stuff in a subclass.
I would love a point of view.